src/Utils/process-message.ts

```
@@ -304,437 +317,474 @@ const processMessage = async (
-					// TODO: investigate
-					if (histNotification.syncType !== proto.HistorySync.HistorySyncType.ON_DEMAND) {
-						ev.emit('creds.update', {
-							processedHistoryMessages: [
-								...(creds.processedHistoryMessages || []),
-								{ key: message.key, messageTimestamp: message.messageTimestamp }
-							]
-						})
-					}
+			switch (protocolMsg.type) {
+				case proto.Message.ProtocolMessage.Type.HISTORY_SYNC_NOTIFICATION:
+					const histNotification = protocolMsg.historySyncNotification!
```

Biome (2.4.15)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)